### ISSUE TITLE
Crashlytics does not start with .NET 8

### ISSUE THREAD

**Issue by cagriy** [TobiasBuchholz/Plugin.Firebase#239]
I've recently upgraded a project where Crashlytics was working fine.

Now when I run the project I am getting following errors:

```
[I-CLS000000] Crashlytics could not find the symbol for the app's main function and cannot start up. This can happen when Exported Symbols File is set in Build Settings. To resolve this, add "__mh_execute_header" as a newline to your Exported Symbols File.

[I-CLS000000] Crash reporting could not be initialized
```

I double checked that strip symbols option was not enabled also tried adding ` <DebugType>full</DebugType>` but still the same issue.

Any idea on what might be the issue ?

**Comment by juniorsaraviao** [comment]
I see the same behavior (but not the errors) using `Plugin.Firebase.Crashlytics` (v2.0.1) and `NET Maui` (v8.0.100). I'm sending crashes to Crashlytics but the dashboard doesn't report any bugs.

<img src="https://github.com/TobiasBuchholz/Plugin.Firebase/assets/43689290/b9ec5e50-94ab-463d-9198-13f4d6728bd6" width="600" height="300">

when I use `Plugin.Firebase.Crashlytics` (v2.0.1) and `NET Maui` (v7), the dashboard detects the app even if there's no bug reported

<img src="https://github.com/TobiasBuchholz/Plugin.Firebase/assets/43689290/12fed8ea-52fd-4af6-bfec-2f80d43915ec" width="600" height="300">

**Comment by TobiasBuchholz** [comment]
Hi @juniorsaraviao, `Plugin.Crashlytics` is just a really lightweight wrapper around the native firebase crashlytics sdks to make it cross platform, so I don't think your described behavior is caused by this project but by the native sdks. At least I have no idea how to fix that, so my suggestion would be to create an issue at their github repositories: 

- https://github.com/xamarin/GoogleApisForiOSComponents
- https://github.com/xamarin/GooglePlayServicesComponents

**Comment by juniorsaraviao** [comment]
Hi @TobiasBuchholz, I just opened a bug into GoogleApisForiOSComponents [here](https://github.com/xamarin/GoogleApisForiOSComponents/issues/643)

**Comment by kocevilija** [comment]
Did anyone find solution about this problem? I am constantly getting 

```
{Java.Lang.IllegalStateException: The Crashlytics build ID is missing. This occurs when Crashlytics tooling is absent from your app's build configuration. Please review Crashlytics onboarding instructions and ensure you have a valid Crashlytics account.--- …}
```

and cannot get around it.

**Comment by cagriy** [comment]
@kocevilija, I moved on to Sentry in the end. This is a fantastic plugging in overall, but Crashlytics in general turned out to be a bit too complicated for me.

**Comment by cagriy** [comment]
Same here, I use the rest of the plugin but not Crashlytics. And actually I can recall that I also had a hard time removing it completely. Let me try to remember:
1 - Make sure there are no crashlytics references "anywhere", might be a good idea to do a text search through files.
2 - If there is a separate nugget packet for Crashlytics try removing it.
3 - After the above, remove obj and bin folders, restart IDE and do a fresh rebuild of the solution.

I am sorry, I am away from my dev machine atm, just trying to recall what I did.

**Comment by kocevilija** [comment]
@tranb3r my googleService jsons are added. What sould I put in here? Its not really clear

`<string name="com.google.firebase.crashlytics.mapping_file_id">none</string>`

**Comment by tranb3r** [comment]
> @tranb3r my googleService jsons are added. What sould I put in here? Its not really clear
> `<string name="com.google.firebase.crashlytics.mapping_file_id">none</string>`

@kocevilija Did you add this line to your `Platforms/Android/Resources/values/string.xml` file?

**Comment by kocevilija** [comment]
> > @tranb3r my googleService jsons are added. What sould I put in here? Its not really clear
> > `<string name="com.google.firebase.crashlytics.mapping_file_id">none</string>`
> 
> @kocevilija Did you add this line to your `Platforms/Android/Resources/values/string.xml` file?

Yes, I have added it. I have followed https://github.com/TobiasBuchholz/Plugin.Firebase/blob/development/README.md and done all steps as specified. 

Also, on iOS I get the error below. I know I need to put it in info.plist but I cant figure where to find this scheme and what key to use.
```
Name: NSInternalInconsistencyException Reason: Please register custom URL scheme 'app-1-1831545804-ios-7af49d8c0467410c130b69' in the app's Info.plist file.
```

**Comment by mfriese** [comment]
> @tranb3r my googleService jsons are added. What sould I put in here? Its not really clear
> 
> `<string name="com.google.firebase.crashlytics.mapping_file_id">none</string>`

This solved it for me.

**Comment by sylac** [comment]
This solved the issue for me: https://stackoverflow.com/questions/77913820/maui-app-crashing-after-adding-firebase-crashlytics
> Now Important thing to do is:
>
> Right click on string.xml file, and click on 'properties' now you will see a small window where you need to set 'Build Action' as 'Android Resource'.

@TobiasBuchholz I'd propose to add this to crashlitics readme file, for those who needed to create this file.

**Comment by prasad210** [comment]
Any Update on this issue ?

**Comment by AdamEssenmacher** [comment]
It should probably be closed... Crashlytics is known to work as of 3.0.0

**Comment by wynston-champion-actabl** [comment]
This still is not working for me either. I have followed all of the steps listed in https://github.com/TobiasBuchholz/Plugin.Firebase/blob/development/docs/crashlytics.md#android-specifics and am still not getting any indication that crashes are being reported on the dashboard.

![image](https://github.com/user-attachments/assets/71bb49c9-def0-479d-aed9-a7465566527b)

**Comment by rezamohamed** [comment]
I also see this issue with Crashlytics 3.0.0 for iOS, .NET 8.

[Firebase/Crashlytics] Version 10.24.0
10.24.0 - [FirebaseCrashlytics][I-CLS000000] Crashlytics could not find the symbol for the app's main function and cannot start up. This can be resolved 2 ways depending on your setup:
 1. If you have Exported Symbols File set in your Build Settings, add "__mh_execute_header" as a newline in your Exported Symbols File.
 2. If you have -exported_symbols_list in your linker flags, remove it.
10.24.0 - [FirebaseCrashlytics][I-CLS000000] Crash reporting could not be initialized

**Comment by miguelpxkadinha** [comment]
Is there any news on this topic?

I'm currently facing the same issue in  .NET8 Maui

Adding the workaround
<_ExportSymbolsExplicitly>false</_ExportSymbolsExplicitly>
 didn't work

**Comment by claudioredi** [comment]
@miguelpxkadinha I was able to fix it adding this to the iOS project file (take into account that my project is a native Xamarin.iOS project so you might need to make it fit for Maui)

     <ItemGroup>
            <_ReferencesLinkerFlags Include="-u__mh_execute_header" Visible="false" />
            <_CustomLinkFlags Include="-exported_symbols_list" Visible="false" />
            <_CustomLinkFlags Include="symbols.list" Visible="false" />
     </ItemGroup

Where `symbols.list` is a file in the root of my iOS project that contains this

    __mh_execute_header

I'm not 100% sure if this part is required

            <_ReferencesLinkerFlags Include="-u__mh_execute_header" Visible="false" />
            
It was at some point to fix a different similar issue with another symbol

**Comment by miguelpxkadinha** [comment]
@claudioredi 
Thanks you! I was able to fix the issue following your comment.
I had to change it a bit to adapt it to Maui

Like you said, I created a symbols.list (I named it exported_symbols.txt) and added it to the folder Platforms/iOS

`

	<ItemGroup Condition="'$(TargetFramework)' == 'net8.0-ios'">

		<_ReferencesLinkerFlags Include="-u__mh_execute_header" Visible="false" />

		<_CustomLinkFlags Include="-exported_symbols_list" Visible="false" />

		<_CustomLinkFlags Include="$(ProjectDir)Platforms/iOS/exported_symbols.txt" Visible="false" />

	</ItemGroup>
`

**Comment by rolivaresminsait** [comment]
> @claudioredi Thank you! I was able to fix the issue following your comment. I had to change it a bit to adapt it to Maui
> 
> Like you said, I created a symbols.list (I named it exported_symbols.txt) and added it to the folder Platforms/iOS
> 
> ```
> <ItemGroup Condition="'$(TargetFramework)' == 'net8.0-ios'">
>     <_ReferencesLinkerFlags Include="-u__mh_execute_header" Visible="false" />
>     <_CustomLinkFlags Include="-exported_symbols_list" Visible="false" />
>     <_CustomLinkFlags Include="$(ProjectDir)Platforms/iOS/exported_symbols.txt" Visible="false" />
> </ItemGroup>
> ```

Thank you for your comment. I have tried this workaround and it works in debug mode, but when I generate the artifact in release mode I get this error:

```
    Undefined symbols for architecture arm64:
      "﻿__mh_execute_header", referenced from:
         -exported_symbol[s_list] command line option
    ld: symbol(s) not found for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

 Any idea how to fix it?
